frontend/src/pages/SignUp.jsx

```diff
@@ -1,11 +1,77 @@
-const SignUp = () => {
-    return(
-      <div>
-        <p>
-          Sign Up Page
-        </p>
-      </div>
-    )
-  }
-  
-  export default SignUp
\ No newline at end of file
+import React from "react";
+import { Link } from "react-router-dom";
+import { useState } from "react";
+import { useAuth } from "../contexts/AuthContext";
+import { useEffect } from "react";
+import { useNavigate } from "react-router-dom";
+
+export default function SignUp() {
+	const [email, setEmail] = useState("");
+	const [password, setPassword] = useState("");
+	const [confirmPassword, setConfirmPassword] = useState("");
+	const [loading, setLoading] = useState(false);
+	const navigate = useNavigate();
+	const { session, register } = useAuth();
+
+	useEffect(() => {
+		if (session) {
+			navigate("/dashboard");
+		}
+	}, [session]);
+
+	async function handleSubmit(e) {
```

Comment:
Ideally we should have some more checks for email validity, and password requirements (>8 characters, mix of uppercase, lowercase, numbers)

Reply:
- Set default to need >8 characters with the mix of things to have

Reply:
For the domain, instead of filtering for those particular domains, add a more general regex, for example at least [one character]@[at least one character].[domain name] or whatnot. Since some people may sign up with company email addresses for example

Maybe try out this regex: `^[\w-\.]+@([\w-]+\.)+[\w-]{2,4}$`